api/api/log_api.go

```diff
@@ -42,38 +42,34 @@ func (l *LogController) ReadLog(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	res, err := l.LogService.ReadLog(&query)
-	if err != nil {
-		log.Errorf("Error while retrieving log %v", err)
-		InternalServerError(fmt.Sprintf("Error while retrieving log for container %s: %s", query.Name, err)).WriteTo(w)
-		return
-	}
+	podLogs := make(chan service.PodLog)
+	stopCh := make(chan struct{})
 
 	// send status code and content-type
 	w.Header().Set("Content-Type", "plain/text; charset=UTF-8")
 	w.WriteHeader(http.StatusOK)
 
-	// stream the response body
-	defer res.Close()
-	buff := bufio.NewReader(res)
-	for {
-		line, readErr := buff.ReadString('\n')
-		_, writeErr := w.Write([]byte(line))
-		if writeErr != nil {
-			// connection from caller is closed
-			return
-		}
+	go func() {
+		for podLog := range podLogs {
```

Comment:
I think it's cleaner if logservice already merge the pod logs into a single stream.